src/internal_rep.rs

```diff
@@ -386,6 +392,23 @@ impl TypeInfo {
         }
         false
     }
+
+    pub fn get_all_parent_names<'a>(&'a self, types: &'a TypeMap) -> Vec<&'a CascadeString> {
+        let mut ret = Vec::new();
+        for parent in &self.inherits {
+            if !ret.contains(&parent) {
+                ret.push(parent);
+                if let Some(parent_ti) = types.get(parent.as_ref()) {
+                    for name in &parent_ti.get_all_parent_names(types) {
+                        if !ret.contains(name) {
```

Comment:
Because of the recursive call, ret may contain types multiple levels of inheritance up.  Simple example would be:

```
resource foo {}

resource bar inherits foo {}

resource baz inherits foo {}

resource qux inherits bar, baz {}
```

In this case, a call to foo.get_all_parent_names() will yield foo twice, once recursively calling bar, and once recursively calling baz.  Since this is a totally reasonable thing to do, we just want to dedup, rather than error.

Reply:
OK, makes sense. I don't remember if explicitly listing duplicated inherited resources raised an error though.

Reply:
Just tried it out, and it compiled fine.  I'll file an issue for later cleanup.

Reply:
filed #56